openupgrade_scripts/scripts/sale/16.0.1.2/pre-migration.py

```diff
@@ -0,0 +1,77 @@
+from openupgradelib import openupgrade, openupgrade_160
+
+_models_renames = [
+    (
+        "sale.payment.acquirer.onboarding.wizard",
+        "sale.payment.provider.onboarding.wizard",
+    ),
+]
+
+_tables_renames = [
+    (
+        "sale_payment_acquirer_onboarding_wizard",
+        "sale_payment_provider_onboarding_wizard",
+    ),
+]
+
+_renames_xmlids = [
+    (
+        "sale_management.menu_product_attribute_action",
+        "sale.menu_product_attribute_action",
+    ),
+]
+
+
+def _noupdate_switch(env):
+    openupgrade.set_xml_ids_noupdate_value(
+        env, "sale", ["model_sale_order_action_share"], False
+    )
+
+
+def _remove_table_constraints(env):
+    openupgrade.delete_sql_constraint_safely(
+        env, "sale", "sale_order", "date_order_conditional_required"
+    )
+
+
+def _drop_sql_views(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        DROP VIEW IF EXISTS report_all_channels_sales
+        """,
+    )
+
+
+def _create_ir_model_data_sale_default_invoice_email_template(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO ir_model_data (name, res_id, module, model, noupdate)
+        SELECT
+            'default_invoice_email_template',
+            ir_config_parameter.id,
+            'sale',
+            'ir.config_parameter',
+            TRUE
+        FROM ir_config_parameter
+        WHERE key = 'sale.default_invoice_email_template'
+        """,
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_models(env.cr, _models_renames)
+    openupgrade.rename_tables(env.cr, _tables_renames)
+    openupgrade.rename_xmlids(env.cr, _renames_xmlids)
+    _drop_sql_views(env)
+    _noupdate_switch(env)
+    _remove_table_constraints(env)
+    openupgrade_160.fill_analytic_distribution(
+        env,
+        "sale_order_line",
+        "account_analytic_tag_sale_order_line_rel",
+        "sale_order_line_id",
+    )
```

Comment:
Hi @duong77476-viindoo, thinking about another solution: why not creating a temporary column `analytic_account_id` on `sale_order_line`, as the analytic account of the SOL is the one of the related SO? This would allow to still use this openupgrade method.

Reply:
@marielejeune Never mind, i restore back to inline implement not using the `openupgrade_160` any more, please retest if needed

Reply:
It works, thanks!